projects/dvl-fw/src/lib/plugins/default/csv-template-project.ts

```diff
@@ -98,36 +106,72 @@ export class CSVTemplateProject extends DefaultProject {
   }
 
   getGraphicVariables(): GraphicVariable[] {
-
     // Setup some default _naive_ graphic variable mappings.
+    const filteredFields = this.getFilteredFields(this.fields);
+    const mappingFields = this.fields.filter((f: string) => !filteredFields.includes(f));
+
+    const mappings = { ...this.getNaiveMappings(filteredFields) };
+    const predefinedMappings = this.getPredefinedMappings(mappingFields);
+
+    Object.entries(predefinedMappings).forEach((entry: [string, {}]) => {
+      if (entry[1]) {
+        mappings[entry[0]][Object.keys(entry[1])[0]] = Object.values(entry[1])[0];
+      }
+    });
+
+    return DefaultGraphicVariableMapping.fromJSON([
+      {
+        recordStream: 'csvData',
+        mappings: {
+          csvData: mappings
+        }
+      }
+    ], this);
+  }
+
+  getPredefinedMappings(mappingFields: string[]): {} {
+    const predefinedMappings = {};
+
+    for (const field of mappingFields) {
+      const splitFields = field.trim().split('$$');
+      if (splitFields.length > 1) {
+        if (!predefinedMappings[splitFields[0]]) {
+          predefinedMappings[splitFields[0]] = {};
+        }
+        predefinedMappings[splitFields[0]][splitFields[1]] = [{selector: splitFields[0]}];
```

Comment:
Have you tested this? It looks wrong. I believe selector: should be `selector: field`.

Reply:
Yes. The 'field' would be of type 'dv$$gv' and splitField[0] would be 'dv'.